js/modules/timeline.src.js

```diff
@@ -0,0 +1,431 @@
+    },
+    states: {
+        hover: {
+            lineWidthPlus: 5,
+            halo: {
+                size: 0
+            }
+        }
+    },
+    dataLabels: {
+        enabled: true,
+        allowOverlap: true,
+        backgroundColor: '#fff',
+        format: '<span style="color:{point.color}">● </span>' +
+            '<span style="font-weight: bold;>{point.date}</span><br/>' +
+            '{point.label}',
+        borderWidth: 1,
+        borderColor: 'gray',
+        distance: 100,
+        alternate: true,
+        verticalAlign: 'middle',
+        color: '#000'
+    },
+    marker: {
+        enabledThreshold: 0,
+        symbol: 'square',
+        radius: 10,
+        height: 15
+    }
+}, {
+    alignDataLabel: function () {
+        var series = this,
+            xAxis = series.xAxis,
+            xAxisLen = xAxis.len,
+            points = series.points,
+            point = arguments[0],
+            pointIndex = points.indexOf(point),
+            isFirstOrLast = !pointIndex || pointIndex === points.length - 1,
+            dataLabelsOptions = series.options.dataLabels,
+            alternate = dataLabelsOptions.alternate,
+            multiplier = alternate ?
+            (isFirstOrLast ? 1.5 : 2) :
+            1,
+            dataLabel = arguments[1],
+            padding = dataLabel.padding,
+            dataLabelWidth = dataLabelsOptions.width || (Math.floor(
+                xAxisLen / points.length
+            ) * multiplier - (padding * 2));
+
+        // Adjust data label width to the currently available value.
+        dataLabel.css({
+            width: dataLabelWidth
+        });
+        Series.prototype.alignDataLabel.apply(series, arguments);
+    },
+    getMappedXData: function () {
+        var series = this,
+            xMap = [],
+            pointsLen = series.xData.length,
+            max = series.xAxis.options.max,
+            base = max / pointsLen;
+
+        for (var i = 1; i <= pointsLen; i++) {
+            xMap.push(
+                (base * i) - (base / 2)
+            );
+        }
+
+        return xMap;
+    },
+    distributeDL: function (hasRendered) {
+        var series = this,
+            chart = this.chart,
+            points = series.points,
+            dataLabelsOptions = series.options.dataLabels,
+            options,
+            pointDLOptions,
+            newOptions = {},
+            index,
+            isInverted = chart.inverted,
+            offset,
+            distance = dataLabelsOptions.distance,
+            alternate = dataLabelsOptions.alternate;
+
+        each(points, function (point) {
+            options = point.options;
+            index = points.indexOf(point);
+            pointDLOptions = point.options.dataLabels;
+
+            if (!hasRendered) {
+                point.userDataLabelOptions = merge({}, pointDLOptions);
+            }
+
+            if (alternate) {
+                offset = index % 2 ? -distance : distance;
```

Comment:
the distance is already declared in the line 139. So better is using `if(index % 2) offset = -distance`

Reply:
The `offset` variable is deleted now. Calculate alternating data labels position is optimized a bit.